### ISSUE TITLE
Spike on updating Ansible to latest version

### ISSUE THREAD

**Issue by dacook** [openfoodfoundation/ofn-install#963]
- https://community.openfoodnetwork.org/t/server-upgrades-2024/3263/7

> First, we will do a spike on upgrading ansible/python directly to latest version (1-2hrs). (Probably Maikel).
> The result of the spike is that we will be able to decide on the exact upgrade path (eg upgrade minor increments or all in one go), and provide an estimate.
> 
> Our plan then would be to schedule in the update, so that we can be prepared to upgrade the next round of servers.

Upgrade guides:

* https://docs.ansible.com/ansible/devel/porting_guides/porting_guides.html

**Comment by dacook** [comment]
@mkllnk I forget sorry, did you have a go at this?

**Comment by mkllnk** [comment]
No, I didn't. It keep slipping my mind. But it could be a good task for tomorrow.

**Comment by mkllnk** [comment]
I'm not sure how to proceed. I've invested over two hours now. Initially I didn't get far in two hours. One problem is the actual time it takes to install dependencies and so many random failures. It's not smooth. But once you get into it, it becomes easier. There's just a learning curve about inter-dependencies and version management.

Now I'm in a flow and get results. I reckon I should keep going a little bit until I hit a roadblock.

**Comment by mkllnk** [comment]
I think that I can conclude my spike here.

I opened a pull request with one little step:

* #974

And I explored more updates that update ansible-core:

* Ansible 4.0
* Ansible 5.0

Both updates were quite smooth. I tested CI and deploy to staging. In both versions deprecation warnings came up. They were easy to solve but I think it's worth testing each version individually to not break anything.

Python is on a currently supported version. It's old enough for our current Ansible version but also new enough to allow us all Ansible update. So we shouldn't need to worry about updating Python in this project.

I would still estimate 2 hours per major version of Ansible, just because it takes so long to test. And you need to test to check for deprecations and make sure it works, of course. There are 8 more major versions. That's a total of 16 hours. It's possible to multi-task though while playbooks are running. I definitely think that we should continue this.

**Comment by dacook** [comment]
Great work, this gives us a clearer path forward!

I note that there are generally two releases a year (three in 2021)([ref](https://endoflife.date/ansible)), so I was wondering if it might be a reasonable compromise to upgrade two versions at once. What do you think?

**Comment by mkllnk** [comment]
Ah, that's a great reference. I think that's a good idea. The more updates we do the more confident we could become as well. I would look at the number of commits. If on upgrade needs a lot of fixes then I wouldn't jump another version. But if there's not much work needed then I can go to the next version or even include a third one.

**Comment by dacook** [comment]
Yes, that's a good plan 👍

**Comment by dacook** [comment]
Spike done, and Maikel has found that it's relatively easy to proceed, so he is testing out new versions on the side of other work.
